Makefile

```diff
@@ -72,17 +72,22 @@ models/small: whisper.cpp/models/ggml-small.bin
 	cp whisper.cpp/models/ggml-small.bin models/small
 
 
-
-bin/main: whisper.cpp
+main: whisper.cpp
+	sed -i 's/^CFLAGS   = -I\. -static              -O3/CFLAGS   = -I.              -O3/' whisper.cpp/Makefile
+	sed -i 's;^CXXFLAGS = -I\. -static -I\./examples -O3;CXXFLAGS = -I. -I./examples -O3;' whisper.cpp/Makefile
+	sed -i 's/^LDFLAGS  = -lstdc++ -lm -lc/LDFLAGS  =/' whisper.cpp/Makefile
 	cd whisper.cpp && make clean && make
 	cp whisper.cpp/main bin/main
 
 
-bin/main-musl: whisper.cpp
-	cd whisper.cpp && make clean && make \
-	-E 'CFLAGS += -D_POSIX_SOURCE -D_GNU_SOURCE' \
-	-E 'CXXFLAGS += -D_POSIX_SOURCE -D_GNU_SOURCE'
-	cp whisper.cpp/main bin/main-musl
+# better to do this in a musl system because of issues with libnss static linking on glibc
+static: whisper.cpp
+	sed -i 's/^CFLAGS   = -I\.              -O3/CFLAGS   = -I\. -static              -O3/' whisper.cpp/Makefile
+	sed -i 's;^CXXFLAGS = -I\. -I\./examples -O3;CXXFLAGS = -I. -static -I./examples -O3;' whisper.cpp/Makefile
+	sed -i 's/^LDFLAGS  =$$/LDFLAGS  = -lstdc++ -lm -lc/' whisper.cpp/Makefile
```

Comment:
Whisper's Makefile allows passing in variables, I believe. Not sure if sed is necessary

Reply:
Agreed, this sed is very fragile. The Makefile could change and we wouldn't notice that the replacement doesn't work anymore

Reply:
yeah I tried that but it didn't work. There was an option to convert those lines to use `override` directive like `override CFLAGS` but that would use sed again.

Reply:
> Agreed, this sed is very fragile. The Makefile could change and we wouldn't notice that the replacement doesn't work anymore

Using a specific commit for the whisper.cpp repo should fix that

Reply:
> Using a specific commit for the whisper.cpp repo should fix that

That should be done anyway, but it could still break when updating.

Reply:
> Can't you override the compiler by doing something like CC="gcc -static" ...?

I was changing CFLAGS and CXXFLAGS all the time. this worked, thanks!